modules/platforms/cpp/ignite/client/detail/table/table_impl.cpp

```diff
@@ -233,7 +268,7 @@ ignite_tuple read_tuple(protocol::reader &reader, const schema *sch, const ignit
         if (i < sch->key_column_count) {
             res.set(column.name, key.get(column.name));
         } else {
-            res.set(column.name, read_next_column(parser, column.type));
+            res.set(column.name, read_next_column(parser, column.type, column.scale));
```

Comment:
Do we have an integration test to check that decimal scale handling in C++ client corresponds to the server? 
* Check roundtrip
* Run a Compute job to compare string representations

Reply:
Roundtrip is checked. I've actually found a bug this way. Compute is not yet supported in C++. I will add an SQL test - something like `SELECT CAST('12345.6789' AS DECIMAL(10, 4))`.